sphinx/ext/napoleon/docstring.py

```diff
@@ -888,6 +1066,12 @@ def _consume_field(self, parse_type: bool = True, prefer_type: bool = False
             _name, _type = line, ''
         _name, _type = _name.strip(), _type.strip()
         _name = self._escape_args_and_kwargs(_name)
+        if self._config.napoleon_use_param:
+            _type = _convert_numpy_type_spec(
+                _type,
+                location=self._get_location(),
+                translations=self._config.napoleon_type_aliases or {},
+            )
```

Comment:
Do you have a plan to support numpy types on `napoleon_use_param` after this PR?

Reply:
this would not be terribly difficult, but we'd need to make sure to reference all types; right now, a parameter definition like
```python
param1 : str
    description
```
would not be changed. However, that could be a bug since now the type aliases wouldn't work. If so, supporting `napoleon_use_param = False` could be as simple as removing the condition.

Reply:
The document of `napoleon_use_param` only says how the options are represented (displayed), and does not say the types are not hyperlinked.
https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#confval-napoleon_use_param

So I think it's an improvement of `napleon_use_param = False` case. It's okay to change test cases.

Reply:
well, if `napoleon_use_param = False`, the type spec is wrapped in `*{}*`, so I guess there's more to do to support that. Let's leave that to a new PR?